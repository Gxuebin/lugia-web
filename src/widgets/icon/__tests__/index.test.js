//@flow

import * as React from 'react';
import chai from 'chai';
import Icon from '../';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { mount, } from 'enzyme';

const { mockFunction, VerifyOrder, VerifyOrderConfig, } = require('vx-mock');

const { expect: exp, } = chai;

describe('Icon', () => {

  it('iconClass: sv-icon-close', () => {
    const target = renderer.create(<Icon iconClass="sv-icon-close"></Icon>).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('iconClass: sv-icon-close', () => {
    const order = VerifyOrder.create();
    const mockClick = mockFunction.create(VerifyOrderConfig.create('eventHandle', order));
    const onClick = mockClick.getFunction();
    const target = mount(<Icon iconClass="sv-icon-close" onClick={onClick}></Icon>);
    target.find('i').simulate('click', {});
    order.verify(obj => {
      const { eventHandle, } = obj;
      eventHandle(VerifyOrder.Object);
    });
  });

});
