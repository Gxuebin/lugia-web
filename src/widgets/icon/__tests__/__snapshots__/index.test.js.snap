// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon iconClass: lugia-icon-reminder_close_circle_o 1`] = `
.c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

<span
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    className="lugia-icon-reminder_close_circle_o   c0"
    style={Object {}}
  />
</span>
`;
